### PR TITLE
EIP-1559 Support

### LINKED ISSUE #35: Support EIP-1559 once approved
Importantly we'll need to know how the RPC payloads change so we can deserialize e.g. `eth_getTransactionByHash` on these txs into `eth.Transaction`.

### PR DESCRIPTION
**Pull request by ryanschneider**
Fixes #35 (finally!)

I wrote a little test script that processes every baikal block and verifies the RLP of the block and the RLP of every tx in every block, so am pretty confident this all works correctly.  Since the script uses some non-open source code I haven't included it here.

### CODE REVIEW

File: eth/block_from_raw.go [hunk 1]
```diff
@@ -33,8 +33,10 @@ func (b *Block) FromRaw(input string) error {
 	hash, err := NewHash(h)
 
 	header, txs, uncles := decoded.List[0].List, decoded.List[1].List, decoded.List[2].List
-	// header should be 15 items
-	if len(header) != 15 {
+	// header should be 15 items for legacy blocks, 16 for EIP-1559 blocks
+	switch len(header) {
+	case 15, 16:
```
Comment (kshinn) [inline eth/block_from_raw.go:38]: I've not seen this construct used like this before. Does this short hand a no-op for `case 15, 16` and it does not fall through like other languages?

**Review by kshinn — APPROVED** [review]

File: eth/block_from_raw.go (hunk 1 above)
Comment (ryanschneider) [reply eth/block_from_raw.go:38]: Ya Go has an explicit `fallthrough` keyword, see for example: https://play.golang.org/p/p_QIKattOL

File: eth/transaction.go [hunk 2]
```diff
@@ -28,6 +33,13 @@ type Transaction struct {
 	R           Quantity  `json:"r"`
 	S           Quantity  `json:"s"`
 
+	// Gas Price (optional since not included in EIP-1559)
+	GasPrice *Quantity `json:"gasPrice,omitempty"`
```
Comment (JeeChoi) [inline eth/transaction.go:37]: making sure I understand this correctly for TLM, but it looks like GasPrice is "going away" and being replaced with MaxFeePerGas & MaxPriorityFeePerGas for EIP-1559. Is there a case we would get all 3 values in a txs?

**Review by JeeChoi — APPROVED** [review]
lgtm

File: eth/transaction.go (hunk 2 above)
Comment (ryanschneider) [reply eth/transaction.go:37]: Nope, it should be either GasPrice alone for legacy txs or `MaxFeePerGas` and `MaxPriorityFeePerGas` for -1559 txs.

**Review by egalano — APPROVED** [review]
Went through this in detail. Looks good based on comparing it to the expected changes in the EIP. :shipit:

**Pull request merged**